logic/resolvable/Concludable.java

```diff
@@ -249,6 +249,12 @@ else if (a.getType().isDouble())
         return predicateFn;
     }
 
+    protected void addValueEqualsRequirements(Unifier.Builder unifierBuilder, Set<ValueConstraint<?>> values) {
+        for (ValueConstraint<?> value : equalsConstraints(values)) {
```

Comment:
maybe able to inline `equalsConstraints` if not used anywhere else

Reply:
It's used in 3 places which is why I pulled it out: in Has, Value and Attribute.